modules/nixos/graphics/amd.nix

```
@@ -0,0 +1,17 @@
+{ lib, config, ... }:
+let
+  facterLib = import ../../../lib/lib.nix lib;
+  cfg = config.facter.detected.graphics.amd;
+in
+{
+  options.facter.detected.graphics = {
+    amd.enable = lib.mkEnableOption "Enable the AMD Graphics module" // {
+      default = builtins.elem "amdgpu" (
+        facterLib.collectDrivers (config.facter.report.hardware.graphics_card or [ ])
+      );
+    };
+  };
+  config = lib.mkIf cfg.enable {
+    services.xserver.videoDrivers = [ "amdgpu" ];
```

Thoughts about that?

```suggestion
    services.xserver.videoDrivers = [ "amdgpu" ];
    hardware.amdgpu.initrd.enable = lib.mkDefault true;
```

---

We might need to card this like:

```
      if lib.versionOlder lib.version "24.11pre" then
```

Because that option doesn't exit in 24.05.

---

Ah wait, we already doing this in the main module.